pkg/kapp/resources/resources.go

```diff
@@ -96,6 +96,21 @@ func (c *ResourcesImpl) All(resTypes []ResourceType, opts AllOpts) ([]Resource,
 		opts.ListOpts = &metav1.ListOptions{}
 	}
 
+	// TODO: Move to a separate function
+	if len(opts.ResourceNamespaces) > 0 {
+		exists := map[string]bool{}
+		for _, ns := range c.opts.FallbackAllowedNamespaces {
+			exists[ns] = true
+		}
+		// Populate FallbackAllowedNamespace with resource namespaces stored during deploy
+		for _, ns := range opts.ResourceNamespaces {
+			if !exists[ns] && ns != "(cluster)" {
+				c.opts.FallbackAllowedNamespaces = append(c.opts.FallbackAllowedNamespaces, ns)
```

Comment:
I have added the resource namespaces to the existing fallback namespaces, but I was thinking that we can actually replace it with the resource namespaces because if there is a resource in the fallback Namespace, it would already be present in the resource namespace. 
(Although it doesn't hurt to have one extra namespace)